WordPress/gutenberg
# Button Block: Update to use color support utils to retrieve classes and styles

## Description

This updates the button block to use a [new utility exported from the block support hooks](https://github.com/WordPress/gutenberg/pull/30869) to retrieve color related CSS classes and inline styles which are then applied to inner elements.

The original `color-props.js` function has been re-worked into the new utilities mentioned above.

## How has this been tested?

Manually and via `npm run test-unit` to ensure fixture tests still show the correct classes and styles being applied.

#### Test Instructions

1. Create a new post and add a button block to it
2. Adjust the button's colors via the inspector controls and check the button updates appropriately
3. Save the post and confirm the colors are correct on the frontend.
4. Run `npm run test-unit` and confirm no failures.

## Types of changes

Refactor. No new features, no breaking changes.

## Checklist:
